src/main/java/cz/kominekjan/disenchantment/config/Config.java

```diff
@@ -37,15 +36,111 @@ public static List<Material> getDisabledMaterials() {
         return new ArrayList<>(config.getStringList(ConfigKeys.DISABLED_ITEMS.getKey()).stream().map(Material::getMaterial).toList());
     }
 
-    public static Map<Enchantment, Boolean> getDisabledEnchantments() {
-        List<String> list = config.getStringList(ConfigKeys.DISABLED_ENCHANTMENTS.getKey());
-        return list.stream().map(s -> {
-            String[] split = s.split(":");
-            return Map.entry(
-                    Objects.requireNonNull(Registry.ENCHANTMENT.stream().filter(e -> e.getKey().getKey().equals(split[0])).findFirst().orElse(null)),
-                    Boolean.parseBoolean(split[1])
-            );
-        }).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    public static Map<Enchantment, EnchantmentStatus> getBookSplittingEnchantmentsStatus(){
+        return getGeneralEnchantmentStatus(ConfigKeys.BOOK_SPLITTING_ENCHANTMENTS_STATUS, ConfigKeys.DISABLED_BOOK_SPLITTING_ENCHANTMENTS);
+    }
+
+    public static Map<Enchantment, EnchantmentStatus> getEnchantmentsStatus(){
+        return getGeneralEnchantmentStatus(ConfigKeys.ENCHANTMENTS_STATUS, ConfigKeys.DISABLED_ENCHANTMENTS);
+    }
+
+    private static Map<Enchantment, EnchantmentStatus> getGeneralEnchantmentStatus(ConfigKeys newKey, ConfigKeys legacyKey){
+        Map<Enchantment, EnchantmentStatus> statuses = new HashMap<>();
+        ConfigurationSection section = config.getConfigurationSection(newKey.getKey());
+        if(section == null) {
+            fillLegacyStatus(legacyKey, statuses);
+            return statuses;
+        }
+
+        for (String enchantmentKey : section.getKeys(false)) {
+            if(!section.isString(enchantmentKey)) continue;
+
+            EnchantmentStatus status = EnchantmentStatus.getStatusByName(section.getString(enchantmentKey));
+            if(status == null) continue;
+
+            Registry.ENCHANTMENT
+                    .stream()
+                    .filter(e -> e.getKey().getKey().equals(enchantmentKey))
+                    .findFirst()
+                    .ifPresent(enchantment -> statuses.put(enchantment, status));
+        }
+
+        fillLegacyStatus(legacyKey, statuses);
+        return statuses;
+    }
+
+    // legacy (disabled-enchantment) compatibility
+    private static void fillLegacyStatus(ConfigKeys legacyKey, Map<Enchantment, EnchantmentStatus> statuses) {
+        ConfigurationSection legacySection = config.getConfigurationSection(legacyKey.getKey());
+        if (legacySection == null) return;
+
+        for (String enchantmentKey : legacySection.getKeys(false)) {
+            EnchantmentStatus status;
+            if(!legacySection.isBoolean(enchantmentKey)){
+                // Not "false" or "true". we default to enabled
+                status = EnchantmentStatus.ENABLED;
```

Comment:
enabled should be default and should not be written to a config file, takes up space, every byte counts! :DDD

Reply:
I do not understand why you commented for this line. But I will do it for the write function!

Reply:
Resolved on https://github.com/H7KZ/Disenchantment/pull/15/commits/be329a3a35cfd14a12c38932cd9332b56b2ba090